libkbfs/prefetcher.go

```diff
@@ -272,23 +273,36 @@ func (p *blockPrefetcher) recordPrefetchParent(childBlockID kbfsblock.ID,
 		p.prefetches[childBlockID] = pre
 		needNewFetch = true
 	}
-	if !pre.parents[parentBlockID] {
-		// The new parent needs its subtree block count increased.
+	_, isParentWaiting := p.prefetches[parentBlockID]
+	if !isParentWaiting {
+		p.log.CDebugf(pre.ctx, "prefetcher doesn't know about parent block "+
+			"%s for child block %s", parentBlockID, childBlockID)
+		panic("prefetcher doesn't know about parent block when trying to " +
+			"record parent-child relationship")
+	}
+	if !pre.parents[parentBlockID] || isParentNew {
+		// The new parent needs its subtree block count increased.  This
+		// happens when either the child didn't know about the parent, or the
+		// parent was newly created.
```

Comment:
Please elaborate here why both of these cases could happen, since I don't think it's obvious.